components/mdns/mdns.c

```diff
@@ -6428,16 +6550,55 @@ esp_err_t mdns_service_subtype_update_multiple_items_for_host(const char *instan
 {
     MDNS_SERVICE_LOCK();
     esp_err_t ret = ESP_OK;
-    int cur_index = 0;
+    
     ESP_GOTO_ON_FALSE(_mdns_server && _mdns_server->services && !_str_null_or_empty(service_type) && !_str_null_or_empty(proto),
                       ESP_ERR_INVALID_ARG, err, TAG, "Invalid state or arguments");
 
     mdns_srv_item_t *s = _mdns_get_service_item_instance(instance_name, service_type, proto, hostname);
     ESP_GOTO_ON_FALSE(s, ESP_ERR_NOT_FOUND, err, TAG, "Service doesn't exist");
 
-    // TODO: find subtype needs to say sendbye
+    mdns_subtype_t *srv_subtype = s->service->subtype;
```

Comment:
Nits: Let's define a function for finding the subtype list which needs to say sendbye.